wallet/wallet.go

```diff
@@ -823,9 +826,14 @@ func (w *Wallet) FetchHeaders(chainClient *chain.RPCClient) (count int,
 		if err != nil {
 			return
 		}
+	} else {
```

Comment:
instead of doing this just make a single call to w.TxStore.MainChainTip below this if block.